third_party/terraform/tests/resource_compute_router_nat_test.go

```diff
@@ -188,9 +223,107 @@ func testAccComputeRouterNatBasic(testId string) string {
 	`, testId, testId, testId, testId)
 }
 
+// Like basic but with extra resources
+func testAccComputeRouterNatBasicBeforeUpdate(randPrefix string) string {
+	return fmt.Sprintf(`
```

Comment:
nit: since we're using a preformatted string, we can left-align the HCL instead of indenting it as if it were Go code.